ApplicationLibCode/FileInterface/RifRoffReader.cpp

```diff
@@ -21,121 +21,48 @@
 #include <QFile>
 #include <QTextStream>
 
-const QString codeValuesString = QString( "array int codeValues" );
-const QString codeNamesString  = QString( "array char codeNames" );
-const QString headerString     = QString( "roff-asc" );
+#include "roffcpp/src/Reader.hpp"
 
-bool RifRoffReader::isCodeValuesDefinition( const QString& line )
-{
-    return line.startsWith( codeValuesString );
-}
-
-bool RifRoffReader::isCodeNamesDefinition( const QString& line )
-{
-    return line.startsWith( codeNamesString );
-}
-
-bool RifRoffReader::isCorrectHeader( const QString& line )
-{
-    return line.startsWith( headerString );
-}
-
-int RifRoffReader::extractNumberAfterString( const QString& line, const QString& prefix )
-{
-    QString copiedLine( line );
-
-    bool ok;
-    int  num = copiedLine.remove( prefix ).toInt( &ok );
-    if ( !ok )
-    {
-        throw RifRoffReaderException( "Unexpected value: not an integer." );
-    }
-    return num;
-}
+#include <fstream>
 
-int RifRoffReader::extractCodeValuesCount( const QString& line )
+void RifRoffReader::readCodeNames( const QString& filename, const QString& parameterTagName, std::map<int, QString>& codeNames )
 {
-    return extractNumberAfterString( line, codeValuesString );
-}
+    codeNames.clear();
 
-int RifRoffReader::extractCodeNamesCount( const QString& line )
-{
-    return extractNumberAfterString( line, codeNamesString );
-}
-
-void RifRoffReader::readCodeNames( const QString& filename, std::map<int, QString>& codeNames )
-{
-    QFile file( filename );
-    if ( !file.open( QIODevice::ReadOnly | QIODevice::Text ) )
+    std::ifstream stream( filename.toStdString(), std::ios::binary );
```

Comment:
The file type is supposed to be text files

Reply:
@magnesj Yes, the input files are roff ascii, but the roff::Reader needs to get a binary stream of that data in order to work. I don't remember the details, but I think it was related to istream::read and handling of newline. I think you should leave it as it is, @jorgenherje.